debian/patches/gflag-time-limit.diff

```diff
@@ -0,0 +1,139 @@
+--- a/tcpdump.c
++++ b/tcpdump.c
+@@ -211,6 +211,8 @@
+  */
+ extern int dflag;
+ int dflag;				/* print filter code */
++static int gflag;			/* exit after this many seconds */
++static time_t gflag_time;		/* time_t to exit at */
+ static int Gflag;			/* rotate dump files after this many seconds */
+ static int Gflag_count;			/* number of files created with Gflag rotation */
+ static time_t Gflag_time;		/* The last time_t the dump file was rotated. */
+@@ -278,8 +280,10 @@
+ #ifdef _WIN32
+     static HANDLE timer_handle = INVALID_HANDLE_VALUE;
+     static void CALLBACK verbose_stats_dump(PVOID param, BOOLEAN timer_fired);
++    static void CALLBACK gflag_timer_fired(PVOID param, BOOLEAN timer_fired);
+ #else /* _WIN32 */
+   static void verbose_stats_dump(int sig);
++  static void gflag_timer_fired(int sig);
+ #endif /* _WIN32 */
+ 
+ static void info(int);
+@@ -672,7 +676,7 @@
+ #define U_FLAG
+ #endif
+ 
+-#define SHORTOPTS "aAb" B_FLAG "c:C:d" D_FLAG "eE:fF:G:hHi:" I_FLAG j_FLAG J_FLAG "KlLm:M:nNOpq" Q_FLAG "r:s:StT:u" U_FLAG "vV:w:W:xXy:Yz:Z:#"
++#define SHORTOPTS "aAb" B_FLAG "c:C:d" D_FLAG "eE:fF:g:G:hHi:" I_FLAG j_FLAG J_FLAG "KlLm:M:nNOpq" Q_FLAG "r:s:StT:u" U_FLAG "vV:w:W:xXy:Yz:Z:#"
+ 
+ /*
+  * Long options.
+@@ -1631,13 +1635,27 @@
+ 			infile = optarg;
+ 			break;
+ 
++		case 'g':
++			gflag = atoi(optarg);
++			if (gflag < 1)
++				error("invalid number of seconds %s", optarg);
++
++			/* Grab the current time for time limit. */
++			if ((gflag_time = time(NULL)) == (time_t)-1) {
```

Comment:
Do we care that time can go backwards if NTP adjusts it aggressively?

Reply:
It is a plausible scenario, however not one that I think needs worrying about: The existing periodic config (-G) would also be affected here.

Reply:
Fine. Meh.